extensions/mssql/src/features.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { SqlOpsFeature, SqlOpsDataClient } from 'dataprotocol-client';
+import { ClientCapabilities, ServerCapabilities, RPCMessageType, StaticFeature } from 'vscode-languageclient';
+
+import { Disposable } from 'vscode';
+
+import { Telemetry } from './telemetry';
+import * as Utils from './utils';
+import { TelemetryNotification } from './contracts';
+
+class TelemetryFeature implements StaticFeature {
+
+	constructor(private _client: SqlOpsDataClient) { }
+
+	fillClientCapabilities(capabilities: ClientCapabilities): void {
+		Utils.ensure(capabilities, 'telemetry')!.telemetry = true;
+	}
+
+	initialize(): void {
+		const client = this._client;
```

Comment:
Should we decide to keep this class, `client` isn't necessary. We can just reference `this._client`

Reply:
there are closure issues with doing that. assigning client is a alternative to assigning this to self. This is achieving the same thing

Reply:
I don't think there's any closure issues here since `client` isn't being used inside the onNotification handler. Or am I fundamentally missing something here?